File: testdata/java/static_field_with_constructor_init.java
class Test {
    // Single constructor with matching params - should work cleanly
    public static final Test INSTANCE = new Test(42, "example");

    // Test with multiple overloaded constructors (ambiguous by count)
    public static final Test AMBIGUOUS = new Test(0, 0, 0);

    private int value;
    private String name;

    public Test(int value, String name) {
        this.value = value;
        this.name = name;
    }

    // Create ambiguity: multiple constructors with 3 params
    public Test(int a, int b, int c) {
        this.value = a;
    }

    public Test(int x, String y, int z) {
        this.value = x;
        this.name = y;
    }
}

File: testdata/go/static_field_with_constructor_init.go
package converted

type test struct {
	value int
	name  string
}

var INSTANCE = NewTestFromIntString(42, "example")

// FIXME: more than one possible constructor for Test
var AMBIGUOUS = NewTestFromIntIntInt(0, 0, 0)

func NewTestFromIntString(value int, name string) test {
	this := test{}
	this.value = value
	this.name = name
	return this
}

func NewTestFromIntIntInt(a int, b int, c int) test {
	this := test{}
	this.value = a
	return this
}

func NewTestFromIntStringInt(x int, y string, z int) test {
	this := test{}
	this.value = x
	this.name = y
	return this
}
